core/dbt/helper_types.py

```diff
@@ -38,12 +37,6 @@ def json_schema(self) -> JsonDict:
         return {'type': 'number'}
 
 
-class RealEncoder(FieldEncoder):
-    @property
-    def json_schema(self):
-        return {'type': 'number'}
-
-
```

Comment:
it turns out that you can just use `float` everywhere we used `Real`, mypy has special handling for the idea that `int` works for `float` fields.

Reply:
Does this have anything to do with JSON serialization? I remember an issue that popped up where some lib (`snowflake-connector-python` or `agate`, I'm not certain) used a weird `decimal` type that messed with json serialization or pickling or something like that.

Reply:
`agate` (and probably `snowflake-connector-python` too) returns `Decimal` all over. We handle that by having a special JSON encoder that knows about `Decimal`s and converts them to floats. I spent some time on it and this didn't seem to cause any problems on the integration test level - if encoding Decimals were an issue with this change, I'd expect the RPC tests to fail.

Reply:
Ok - the thing that I remember in particular was that there was a strange, different, decimal type (`cdecimal`) that would show up when dbt was via conda, or something crazy like that. Check out this issue: https://github.com/fishtown-analytics/dbt/issues/1155

All good if not action is required, but did want to surface this one as possibly thorny

Reply:
Yeah, the code that handles `Decimal` serialization (our custom `dbt.utils.JSONEncoder`) knows about `cdecimal`. So I don't think any action is required.